app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,113 @@
+class TasksController < ApplicationController
+  
+  
+  def index
+    @greeting = "HELLO, THERE!"
+    @tasks = Task.all
+  end
+  
+  
+  def show
+    task_id = params[:id].to_i
+    @task = Task.find_by(id: task_id)
+    if @task.nil?
+      flash[:error] = "Could not find task with id: " + task_id.to_s
+      redirect_to root_path
+      return
+    end
+  end
+  
+  
+  def new
+    @task = Task.new
+    @task.completion_date = nil
+  end
+  
+  
+  def create
+    @task = Task.new( task_params )
+    
+    @task.save
+    
+    if @task.save
+      redirect_to task_path(@task.id)
+    else
+      render new_task_path
+    end
+  end
+  
+  
+  def edit
+    task_id = params[:id].to_i
+    @edit_task = Task.find_by(id: task_id)
+    
+    if @edit_task.nil?
+      redirect_to root_path
+      return
+    end    
+  end  
+  
+  
+  def update
+    @task = Task.find_by(id: params[:id])
+    if @task.nil?
+      redirect_to root_path
+      return
+    end
+    
+    @task.update( task_params )
+    
+    if @task.update( task_params )
```

Comment:
Same as above-- we see a double update, which again, shouldn't be a problem, but probably isn't what you were aiming for